management/src/main/java/org/ehcache/management/registry/DefaultCollectorService.java

```diff
@@ -0,0 +1,215 @@
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicLong;
+
+import static org.ehcache.internal.executor.ExecutorUtil.shutdownNow;
+
+
+/**
+ * @author Mathieu Carbou
+ */
+@ServiceDependencies({CacheManagerProviderService.class, ManagementRegistryService.class, ExecutionService.class, TimeSourceService.class})
+public class DefaultCollectorService implements CollectorService, CacheManagerListener {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(DefaultCollectorService.class);
+
+  private final AtomicBoolean collectingStats = new AtomicBoolean();
+  private volatile ScheduledFuture<?> task;
+
+  private final ConcurrentMap<String, StatisticQuery.Builder> selectedStatsPerCapability = new ConcurrentHashMap<String, StatisticQuery.Builder>();
+  private final MessageConsumer messageConsumer;
+
+  private volatile TimeSourceService timeSource;
+  private volatile ManagementRegistryService managementRegistry;
+  private volatile ScheduledExecutorService scheduledExecutorService;
+  private volatile CacheManagerProviderService cacheManagerProviderService;
+  private volatile ManagementRegistryServiceConfiguration configuration;
+
+  public DefaultCollectorService(MessageConsumer messageConsumer) {
+    this.messageConsumer = messageConsumer;
+  }
+
+  @Override
+  public void start(ServiceProvider serviceProvider) {
+    timeSource = serviceProvider.getService(TimeSourceService.class);
+    managementRegistry = serviceProvider.getService(ManagementRegistryService.class);
+    configuration = managementRegistry.getConfiguration();
+    cacheManagerProviderService = serviceProvider.getService(CacheManagerProviderService.class);
+    scheduledExecutorService = serviceProvider.getService(ExecutionService.class).getScheduledExecutor(DefaultCollectorService.class.getSimpleName());
+
+    cacheManagerProviderService.getCacheManager().registerListener(this);
+  }
+
+  @Override
+  public void stop() {
+    stopStatisticCollector();
+    shutdownNow(scheduledExecutorService);
+  }
+
+  @Override
+  public void cacheAdded(String alias, Cache<?, ?> cache) {
+    messageConsumer.accept(new DefaultMessage(
+        timeSource.getTimeSource().getTimeMillis(),
+        new ContextualNotification(
+            configuration.getContext().with("cacheName", alias),
+            EhcacheNotification.CACHE_ADDED.name())));
+  }
+
+  @Override
+  public void cacheRemoved(String alias, Cache<?, ?> cache) {
+    messageConsumer.accept(new DefaultMessage(
+        timeSource.getTimeSource().getTimeMillis(),
+        new ContextualNotification(
+            configuration.getContext().with("cacheName", alias),
+            EhcacheNotification.CACHE_REMOVED.name())));
+  }
+
+  @Override
+  @SuppressFBWarnings("SF_SWITCH_NO_DEFAULT")
```

Comment:
We should add no-op cases for the 'ignored' transitions, and an AssertionError throw on default.